### PR TITLE
schema: add support OpsGenie notifier

### PR DESCRIPTION
**Pull request by bobheadxi**
~This adds support for notifiers that:~

~(have schema in https://grafana.com/docs/grafana/latest/administration/provisioning/#alert-notification-channels) 
AND
( (have docs in https://grafana.com/docs/grafana/latest/alerting/notifications/#list-of-supported-notifiers)
OR (have docs on service website, e.g. OpsGenie) )~

~There are surprisingly few documented notifiers - not sure if I should go ahead and add the rest anyway (I personally lean towards a need-to-have basis), since some are rather obscure~

**update** - PR now only adds OpsGenie

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -1072,6 +1072,59 @@
...
+        "type": {
+          "type": "string",
+          "const": "kafka"
+        },
+        "kafkaRestProxy": { "type": "string" },
+        "kafkaTopic": { "type": "string" }
+      }
+    },
+    "GrafanaNotifierPrometheus": {
+      "description": "Prometheus alertmanager notifier - see https://grafana.com/docs/grafana/latest/alerting/notifications/#prometheus-alertmanager",
```
Comment (bobheadxi) [inline schema/site.schema.json:1090]: I just noticed the docs link recommend using a `Webhook` notifier, but this schema is provided in the provisioning docs https://grafana.com/docs/grafana/latest/administration/provisioning/#alert-notification-prometheus-alertmanager

🤦

**Comment by tsenart** [comment]
Where can I read more about the over-arching direction we're going in around this work? I find it hard to reason about the rational for extending our site configuration with such Grafana specific settings.

**Comment by bobheadxi** [comment]
@tsenart this is a follow-up to work on #10641, where @slimsag wrote a nice tl;dr here: https://github.com/sourcegraph/sourcegraph/issues/10641#issuecomment-636212006

> I find it hard to reason about the rational for extending our site configuration with such Grafana specific settings

Yeah, I do feel like we shouldn't have the full set of Grafana notifiers in here, and maybe just support a subset of the most common ones (ie I'd even go so far as *only* include the ones we have now + OpsGenie, and drop the other ones I added here)

**Comment by tsenart** [comment]
@bobheadxi: Thank you. Yeah, regarding this particular change, I agree we should start with a minimal set of supported notifiers unless there's customer need for all of these.

**Comment by slimsag** [comment]
> Yeah, I do feel like we shouldn't have the full set of Grafana notifiers in here, and maybe just support a subset of the most common ones (ie I'd even go so far as only include the ones we have now + OpsGenie, and drop the other ones I added here)

SGTM, thanks for owning this @bobheadxi !

> Where can I read more about the over-arching direction we're going in around this work?

The comment [robert linked to](https://github.com/sourcegraph/sourcegraph/issues/10641#issuecomment-636212006) is indeed a good way to get insight into this specific change (of "grafana in our site config"), for more broad insight into the direction of monitoring and distribution as a whole I plan to put together something I'll be able to share very soon.

(side note -> most of our monitoring stack ideas came before our RFC process and were documented in lengthy GitHub issues, but work on it was (ironically) delayed many months until we did have an RFC process in place. In retrospect I would've liked to have had the time to pull more of this into proper RFC documents, there were just simply more pressing issues at the time. I hope to remedy the situation anyway)

**Pull request merged**